citations-impl/impl/src/test/org/sakaiproject/citation/impl/BasicCitationService.java

```diff
@@ -264,6 +256,62 @@ public void saveCollection(CitationCollection collection)
         }
 
 		/* (non-Javadoc)
+		* @see org.sakaiproject.citation.impl.BaseCitationService.Storage#saveSection(org.sakaiproject.citation.api.CitationCollectionOrder)
+		*/
+		public void saveSection(CitationCollectionOrder citationCollectionOrder)
+		{
+			this.m_citationCollections.put(citationCollectionOrder.getCollectionId() + ":" + citationCollectionOrder.getLocation() + ":" + citationCollectionOrder.getSectiontype(), citationCollectionOrder);
+		}
+
+		/* (non-Javadoc)
+		* @see org.sakaiproject.citation.impl.BaseCitationService.Storage#saveSubsection(org.sakaiproject.citation.api.CitationCollectionOrder)
+		*/
+		public void saveSubsection(CitationCollectionOrder citationCollectionOrder)
+		{
+			this.m_citationCollections.put(citationCollectionOrder.getCollectionId() + ":" + citationCollectionOrder.getLocation() + ":" + citationCollectionOrder.getSectiontype(), citationCollectionOrder);
+		}
+		/* (non-Javadoc)
+		* @see org.sakaiproject.citation.impl.BaseCitationService.Storage#saveCitationCollectionOrders(java.util.ArrayList, java.lang.String)
+		*/
+		public void saveCitationCollectionOrders(List<CitationCollectionOrder> citationCollectionOrders, String citationCollectionId)
+		{
+			for (CitationCollectionOrder citationCollectionOrder : citationCollectionOrders) {
```

Comment:
Not sure what you mean. This class isn't transactional - it's a class used for tests (in the tests package).  It uses hashtables to stores java objects (as opposed to using the db).